test/framework/src/main/java/org/opensearch/test/telemetry/MockTelemetry.java

```diff
@@ -13,46 +13,22 @@
 import org.opensearch.telemetry.metrics.MetricsTelemetry;
 import org.opensearch.telemetry.tracing.TracingTelemetry;
 import org.opensearch.test.telemetry.tracing.MockTracingTelemetry;
-import org.opensearch.threadpool.ThreadPool;
-
-import java.util.concurrent.TimeUnit;
 
 /**
  * Mock {@link Telemetry} implementation for testing.
  */
 public class MockTelemetry implements Telemetry {
-    private final ThreadPool threadPool;
-
     /**
      * Constructor with settings.
      * @param settings telemetry settings.
      */
     public MockTelemetry(TelemetrySettings settings) {
-        this(settings, null);
-    }
 
-    /**
-     * Constructor with settings.
-     * @param settings telemetry settings.
-     * @param threadPool thread pool to watch for termination
-     */
-    public MockTelemetry(TelemetrySettings settings, ThreadPool threadPool) {
-        this.threadPool = threadPool;
     }
 
     @Override
     public TracingTelemetry getTracingTelemetry() {
-        return new MockTracingTelemetry(() -> {
-            // There could be some asynchronous tasks running that we should await for before the closing
-            // up the tracer instance.
-            if (threadPool != null) {
```

Comment:
Just to reconfirm  we don't need this anymore? (by all means - this is good)

Reply:
Yes, we are shutting down adding the new spans to processor, so should be good.